core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala

```diff
@@ -347,6 +348,8 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
     def writeNextInputToStream(dataOut: DataOutputStream): Boolean
 
     def open(dataOut: DataOutputStream): Unit = Utils.logUncaughtExceptions {
+      val isUnixDomainSock = authHelper.conf.get(PYTHON_UNIX_DOMAIN_SOCKET_ENABLED)
+      lazy val sockPath = new File(System.getProperty("user.home"), s".${UUID.randomUUID()}.sock")
```

Comment:
The reason why I am using home address is that UDS restricts its maximum length to 104 bytes. It's OS level restriction so can't really work around. If you use the current working directory, it becomes a problem in Spark Connect because the current working directory changes for individual session for session isolation in Spark Connect.

Reply:
It's probably something we are able to improve by using the current working directory but would like to leave it as the room for improvement for now in this PR.

Reply:
Just put it at the root of the temp dir? The user home is rather unsavoury.

Reply:
The problem is that the temp directory can be pretty long ... and I have to pass the absolute path around, and it hit the limit. At least it did in my mac.